packages/cli/src/TagHelpers.ts

```diff
@@ -86,3 +88,67 @@ export async function removeRelations(workflowId: string, tablePrefix: string) {
 		.where('workflowId = :id', { id: workflowId })
 		.execute();
 }
+
+const createTag = async (transactionManager: EntityManager, name: string): Promise<TagEntity> => {
+	const tag = new TagEntity();
+	tag.name = name;
+	return transactionManager.save<TagEntity>(tag);
+};
+
+const findOrCreateTag = async (
+	transactionManager: EntityManager,
+	importTag: ITagToImport,
+	tagsEntities: TagEntity[],
+): Promise<TagEntity> => {
+	// Assume tag is identical if createdAt date is the same to preserve a changed tag name
+	const identicalMatch = tagsEntities.find(
+		(existingTag) =>
+			existingTag.id.toString() === importTag.id.toString() &&
+			existingTag.createdAt &&
+			importTag.createdAt &&
+			new Date(existingTag.createdAt) === new Date(importTag.createdAt),
+	);
+	if (identicalMatch) {
+		return identicalMatch;
+	}
+
+	// Find tag with identical name
+	const nameMatch = tagsEntities.find((existingTag) => existingTag.name === importTag.name);
+	if (nameMatch) {
+		return nameMatch;
+	}
+
+	// Create new Tag
+	const createdTag = await createTag(transactionManager, importTag.name);
+	return createdTag;
```

Comment:
1. The comment is unneeded.
2. No need to assign and return, just return the created tag.